### ISSUE TITLE
Trying to get in touch regarding a security issue

### ISSUE THREAD

**Issue by psmoros** [instructure/canvas-lms#2184]
Hello 👋

I run a security community that finds and fixes vulnerabilities in OSS. A researcher (@thewhiteevil) has found a potential issue, which I would be eager to share with you.

Could you add a `SECURITY.md` file with an e-mail address for me to send further details to? GitHub [recommends](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository) a security policy to ensure issues are responsibly disclosed, and it would help direct researchers in the future.

Looking forward to hearing from you 👍

(cc @huntr-helper)

**Comment by jstanley0** [comment]
Thank you for the suggestion. I've forwarded your request to our security team. Meanwhile you can direct your researcher to report the issue to security@instructure.com

**Comment by psmoros** [comment]
Thanks will do :)

**Comment by Thewhiteevil** [comment]
Hi everyone,

Maintainer you have to accept this report through huntr.dev, you need
repository owner level access to read report, until it is private.

If I send mail you solve it here, who will accept or reject in huntr.dev.
If I self close the report I will get negative points.

Hope you understand.

**Comment by psmoros** [comment]
Just sent the email :) @Thewhiteevil self closing no longer penalises your reputation

**Comment by Thewhiteevil** [comment]
Thanks for the update. I will send mail soon.

**Comment by Thewhiteevil** [comment]
TeHope you are doing well on your end. :)

*. While doing some research on the
https://canvas.instructure.com/files?preview=209446732, I found a stored
cross site scripting attack.

*. Attackers can inject malicious code into the affected application, which
will then be executed in the browser of any user who views the affected
page.

*. This vulnerability could be used to steal user data or perform other
malicious actions.

Proof of concept:

*. I have created a video demonstration of the vulnerability and uploaded
it to my Google Drive.

*. The link for the video is provided below for your review:

https://drive.google.com/file/d/12vdhK_TU3Jett_1u84ZMLobzIFNsk-qT/view?usp=sharing
Reproduction Steps:

*. Go to the https://www.instructure.com/

*. Create instructor account.

*. Move on to the https://canvas.instructure.com/files

*. In file upload part, upload the xss.html file.

*. Then click the file preview

*. Next It will start firing the xss pop-up

*. That's it.
XSS.HTML

*. Create a html file using this payload:

" ><img src="x" onerror="prompt(1);">
Impact

*. The impact of a stored cross-site scripting (XSS) vulnerability is that
an attacker can inject malicious code into the affected application, which
will then be executed in the browser of any user who views the affected
page.

*. With this vulnerability, an attacker can steal sensitive user data such
as cookies or personal information. They can also redirect users to
phishing sites or inject malware which can harm the user's device and steal
their personal information. Additionally, an attacker can use this
vulnerability to impersonate the user and perform actions on their behalf,
such as making unauthorised purchases or posting malicious content.

*. Attacker can leverage a stored cross-site scripting (XSS) vulnerability
to perform browser hooking, which can be used to gain access to a user's
browser and steal sensitive information. Tools such as BeEF (Browser
Exploitation Framework) can be used by attackers to perform browser hooking
and steal data such as cookies, login credentials, and other sensitive
information.

*. Once an attacker successfully hooks a browser, they can use it to
perform a variety of malicious actions such as keylogging, form grabbing,
and capturing screenshots. They can also use the browser to launch other
attacks such as phishing, malware delivery, and clickjacking.

*. It is important for organizations to address stored XSS vulnerabilities
as soon as they are discovered to prevent attackers from gaining access to
users' browsers and steal sensitive information. Additionally, it is
important for users to keep their web browsers and plugins updated to
protect against known vulnerabilities that can be exploited by these tools.

*. It is important for organizations to take this vulnerability seriously
and address it as soon as possible to protect their users data and prevent
further harm.
Solution:

*. One solution for preventing stored cross-site scripting (XSS) attacks is
to properly validate and sanitize user input on the server-side before it
is stored.

*. This can include stripping out any potentially dangerous characters or
tags, such as <script> tags.

*. Another solution is to use a Content Security Policy (CSP) that can help
to restrict the types of scripts and other resources that a browser will
execute on a page.

*. Additionally, encoding user input before displaying it on a page can
also help to prevent XSS attacks.

*. Force the users to download files else.
